modules/api/api.py

```diff
@@ -1,14 +1,21 @@
 import time
 import uvicorn
 from gradio.processing_utils import encode_pil_to_base64, decode_base64_to_file, decode_base64_to_image
-from fastapi import APIRouter, Depends, HTTPException
+from fastapi import APIRouter, Depends, FastAPI, HTTPException
 import modules.shared as shared
-from modules import devices
 from modules.api.models import *
 from modules.processing import StableDiffusionProcessingTxt2Img, StableDiffusionProcessingImg2Img, process_images
 from modules.sd_samplers import all_samplers
 from modules.extras import run_extras, run_pnginfo
 
+apisToExtendWith: list[tuple[str, FastAPI]] = list()
+
+def extendApi(mountUrl: str, subApi: FastAPI):
```

Comment:
extendAPI would need to be called prior to the gradio server starting in webui.py - you should think about how this should hook into extensions at that level (on startup)

Reply:
The API is the very last thing to launch, all extensions and scripts are loaded before. Just create your endpoints and call `extendApi` when your script file is loaded and that's all.

For example, in `extensions/test-extension/scripts` create a python file with this code

```python
from fastapi import FastAPI
from modules.api.api import extendApi

subapi = FastAPI()

@subapi.get("/sub")
def read_sub():
    return {"message": "Hello World from sub API"}

extendApi("/subapi", subapi)`
def extendApi(mountUrl: str, subApi: FastAPI):
```

And that all you need, you can now start the server and go to `localhost:7860/subapi/docs` to see the documentation of your new endpoint.

Reply:
looks good to me then